flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -230,6 +231,18 @@ public HighAvailabilityServices getHighAvailabilityServices() {
 		}
 	}
 
+	@VisibleForTesting
+	@Nonnull
+	protected Map<ResourceID, TaskExecutor> getTaskManagers() {
+		return taskManagers;
+	}
+
+	@VisibleForTesting
+	@Nonnull
+	protected Object getManipulationLock() {
+		return lock;
+	}
```

Comment:
This we should definitely not do. This leaks the internal implementation details of this class to its sub classes.